docker/tensorflow-application/video-serving/streaming.py

```diff
@@ -0,0 +1,56 @@
+from flask import Flask, Response, request
+from flask import render_template, send_file
+import argparse
+import datetime
+import imutils
+import time
+import cv2
+import requests
+import numpy as np
+from PIL import Image
+from flask import jsonify
+import time
+from collections import deque
+
+app = Flask(__name__, template_folder='templates')
+
+#Queue implemented for consistent FPS
+q=deque(maxlen=5)
+
+@app.route("/")
+def index():
+	return render_template('index.html')
+
+# TODO Use a Kafka queue which can help us to decouple the two applications 
+# (analysis and streaming) and provide a more scalable approach.
+def stream_to_queue():
+	list = request.json['instances']
+	for l in list:
+		a = np.asarray(l, dtype=np.uint32)
+		b = Image.fromarray(a.astype('uint8')) 
+		c = np.asarray(b)
+		(flag, image) = cv2.imencode(".JPEG", c)
+		q.append(image)
+
+@app.route("/video")
+def video():
+	def produce_frame():
+		while True:
+			if len(q) is 0:
+				continue
+			
+			outputFrame = q.popleft()
+			yield(b'--frame\r\n' b'Content-Type: image/jpeg\r\n\r\n' + 
+				bytearray(outputFrame) + b'\r\n')
+			time.sleep(0.5)
```

Comment:
Did you try shorter sleep times? How did that work out? I still think you can probably reduce this sleep time to something like .25secs. It's not a huge difference processing wise, but should get you a noticeably more "video-like" application.

Reply:
I did try shorter sleep times and the video seemed to go smoother and look better. I've kept it at 0.5 to find a middle ground with the inference time for the batches of frames and to accommodate for that delay. So that it looks more like a continuous stream rather than a burst of images. We can continue to tweak this number later on if you feel it is too big.

Reply:
I'm changing it to 0.25 and we can test it in our cluster to see how it performs.